### PR TITLE
Add multiple labels/aliases or descriptions in one edit

### PR DESCRIPTION
**Pull request by lokal-profil**
Task: [T166871](https://phabricator.wikimedia.org/T166871)

### CODE REVIEW

File: wikidataStuff/WikidataStuff.py [hunk 1]
```diff
@@ -228,37 +253,64 @@ def addLabelOrAlias(self, lang, name, item, summary=None,
...
+    def add_multiple_label_or_alias(self, data, item, case_sensitive=False,
+                                    summary=None):
+        """
+        Add multiple labels or aliases to the item in one edit.
+
+        Adds the name as either a label (if none) or an alias in the
+        given language.
+
+        If adding both an alias and a name in the same language (or multiple
+        aliases) multiple calls to the function are needed.
```
Comment (Vesihiisi) [inline wikidataStuff/WikidataStuff.py]: This looks counterintuitive, but I'm not sure could it could be solved.

**Comment by Vesihiisi** [comment]
General comment: if we try to add a label etc. with an invalid language code, it causes the whole thing to crash trough Pywikibot's OtherPageSaveError. Is this something we want to handle? The list of supported languages is pretty stable so we could check against it.

File: wikidataStuff/WikidataStuff.py [hunk 2]
```diff
@@ -228,37 +253,64 @@ def addLabelOrAlias(self, lang, name, item, summary=None,
...
+
+    def add_multiple_label_or_alias(self, data, item, case_sensitive=False,
+                                    summary=None):
+        """
+        Add multiple labels or aliases to the item in one edit.
+
+        Adds the name as either a label (if none) or an alias in the
+        given language.
+
+        If adding both an alias and a name in the same language (or multiple
```
Comment (lokal-profil) [inline wikidataStuff/WikidataStuff.py]: Fixed through rewrite

File: wikidataStuff/WikidataStuff.py (hunk 1 above)
Comment (lokal-profil) [inline wikidataStuff/WikidataStuff.py]: Fixed

File: wikidataStuff/WikidataStuff.py [hunk 3]
```diff
@@ -228,37 +253,64 @@ def addLabelOrAlias(self, lang, name, item, summary=None,
...
+        aliases) multiple calls to the function are needed.
+
+        @param data: dictionary of language-name pairs
+        @param item: the item to which the label/alias should be added
+        @param summary: summary to append to auto-generated edit summary
+        @param caseSensitive: if the comparison is case sensitive
```
Comment (lokal-profil) [inline wikidataStuff/WikidataStuff.py]: Done

File: wikidataStuff/WikidataStuff.py [hunk 4]
```diff
@@ -228,37 +253,64 @@ def addLabelOrAlias(self, lang, name, item, summary=None,
         @param summary: summary to append to auto-generated edit summary
         @param caseSensitive: if the comparison is case sensitive
         """
-        summary = summary or self.edit_summary
+        data = {lang: name}
+        self.add_multiple_label_or_alias(
+            data, item, case_sensitive=caseSensitive, summary=summary)
+
+    def add_multiple_label_or_alias(self, data, item, case_sensitive=False,
```
Comment (lokal-profil) [inline wikidataStuff/WikidataStuff.py]: But that breaks backwards compatibility!

I've added a note about it and will fix it in a later patch if that is ok (there is a pywikibot decorator for it but something is wrong in how I tried to implement it)

**Comment by lokal-profil** [comment]
With respect to catching bad lang codes. Is this now fetch-able through the api (or pre-loaded in pywikibot)? For a long time it wasn't.

I would say we add it as a later patch if so.

File: wikidataStuff/WikidataStuff.py (hunk 4 above)
Comment (Vesihiisi) [inline wikidataStuff/WikidataStuff.py]: Yeah, it's a good reason. Just kinda makes it awkward using it, not remembering what was spelled how.

**Comment by Vesihiisi** [comment]
Re lng codes, couldn't really find anything in the API, but for the time being just catching the Pywikibot error (`wikibase-api-not-recognized-language`) looks like something that could be done in a separate patch.

File: wikidataStuff/WikidataStuff.py (hunk 4 above)
Comment (Vesihiisi) [inline wikidataStuff/WikidataStuff.py]: Hm, was the decorator problem caused by the typo in the last commit? What's the status now?

**Comment by lokal-profil** [comment]
Follow up tasks:
* https://phabricator.wikimedia.org/T167036
* https://phabricator.wikimedia.org/T167037

**Pull request merged**